### ISSUE TITLE
Invalid Swagger File shows error message dialog on open

### ISSUE THREAD

**Issue by tedepstein** [RepreZen/KaiZen-OpenAPI-Editor#133]
The attached swagger spec immediately shows errors in a modal dialog box when opened.

I expect it to show validation error markers and entries in the problems view, but not to show error dialogs like this.

![image-2016-05-23-17-43-43-186](https://cloud.githubusercontent.com/assets/692813/15801830/2fc78f66-2a6f-11e6-867a-355ef187cd83.png)

**Comment by tedepstein** [comment]
There is some more detail [here](https://modelsolv.atlassian.net/browse/ZEN-2648), including another project with another error shown on attempt to open the file.   I don't think we should attach or reference that project here, but it would be good to check and make sure the fix resolves this problem as well (assuming the fix should apply to that project's error as well.)

**Comment by ghillairet** [comment]
Is it something that should be fixed in SwagEdit, the exception shown in the dialog comes from the parsing of the Swagger model, something that is not done by SwagEdit.

**Comment by andylowry** [comment]
@ghillairet I don't expect we'll be seeing these pop-ups anymore. If I'm correct, these were shown when Live Views attempted to generate their content for malformed swagger models. A couple of recent changes should prevent these situations, including:
1. Code that determined whether a SwagEdit editor contained valid swagger formerly just tried to perform a YAML parse of the editor content. It's now based on the existence of problem markers attached to the edited resource, at ERROR severity level. Live views will show a modal "Please fix your model" message when these markers are detected, and will not attempt generation.
2. Generation of live view content is now guarded against all Exceptions and also StackOverflowError, and catching any of those results in a different modal error message in the live view indicating that a rendering error occurred.

Our only remaining issue now is that sometimes the "rendering error" message appears when the "fix your model" messages should appear (ideally, we should never see the "rendering error" message). There are three known reasons for this:
1. Lack of startup-validation of the editor content (which I believe you're in the process of addressing).
2. Marker change notifications not getting batched when validation is performed. This is related to a change we'd like to make so that whenever the editor state switches between valid and invalid, the live view is immediately refreshed, rather than waiting for the idle timeout to occur. As it stands we see markers cleared before new markers are created during validation, so there's a short period of time when the editor content appears to be valid but in fact is not - leading to likely generation errors. See https://github.com/ModelSolv/RepreZen/pull/1009 for full details, including what I believe will be a very straightforward fix to SwagEdit.
3. Any cases where the editor content is not valid, but this is not detected by SwagEdit validation will probably lead to the "rendering error" message. I imagine we'll stumble on these from time to time, where requirements stated in the specification are not coded in the validation schema. We'll address them as they arise.

So basically, I think we'll be in very good shape with this once we have validation-on-open and and batched marker updates.

**Comment by tfesenko** [comment]
Fixed
